airflow/models/connection.py

```diff
@@ -134,10 +134,32 @@ def __init__(
             self.schema = schema
             self.port = port
             self.extra = extra
+        if self.extra:
+            self._validate_extra(self.extra, self.conn_id)
 
         if self.password:
             mask_secret(self.password)
 
+    @staticmethod
+    def _validate_extra(extra, conn_id):
+        try:
+            _extra_parsed = json.loads(extra)
+            if not isinstance(_extra_parsed, dict):
```

Comment:
```suggestion
            extra_parsed = json.loads(extra)
            if not isinstance(extra_parsed, dict):
```

nit: locally scoped, doesn't need to be private.